src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Services/SyntaxFacts/ISyntaxFacts.cs

```diff
@@ -17,6 +17,64 @@
+    /// both C# and VB have. For example 'IsSimpleName' to correspond to C# and VB's SimpleNameSyntax node.  These 'checking' 
+    /// methods should never fail.
+    /// </item>
+    /// <item>
+    /// 'GetXxxOfYYY' where 'XXX' matches the name of a property on a 'YYY' syntax construct that both C# and VB have.  For
+    /// example 'GetExpressionOfMemberAccessExpression' corresponding to MemberAccessExpressionsyntax.Expression in both C# and
+    /// VB.  These functions should throw if passed a node that the corresponding 'IsYYY' did not return <see langword="true"/> for.
+    /// </item>
+    /// <item>
+    /// 'GetPartsOfXXX(SyntaxNode node, out SyntaxNode/SyntaxToken part1, ...)' where 'XXX' one of the same named Syntax constructs
+    /// that both C# and VB have, and where the returned parts correspond to the members those nodes have in common across the 
+    /// languages.  For example 'GetPartsOfQualifiedName(SyntaxNode node, out SyntaxNode left, out SyntaxToken dotToken, out SyntaxNode right)'
+    /// VB.  These functions should throw if passed a node that the corresponding 'IsXXX' did not return <see langword="true"/> for.
+    /// </item>
+    /// <item>
+    /// Absolutely trivial questions that relate to syntax and can be asked sensibly of each language.  For example,
+    /// if certain constructs (like 'patterns') are supported in that language or not.
+    /// </item>
+    /// </list>
+    /// Importantly avoid:
+    /// <list type="bullet">
+    /// <item>
+    /// Functions that attempt to blur the lines between similar constructs in the same language.  For example, a QualifiedName
+    /// is not the same as a MemberAccessExpression (despite A.B being representable as either depending on context). 
+    /// Features that need to handle both should make it clear that they are doing so, showing that they're doing the right
+    /// thing for the contexts each can arise in (for the above example in 'type' vs 'expression' contexts).
+    /// </item>
+    /// <item>
+    /// Functions which are effectively specific to a single feature are are just trying to find a place to place complex
+    /// feature logic in a place such that it can run over VB or C#.  For example, a function to determine if a position
+    /// is on the 'header' of a node.  a 'header' is a not a well defined syntax concept that can be trivially asked of
+    /// nodes in either language.  It is an excapsulation of a feature (or set of features) level idea that should be in
+    /// its own dedicated service.
+    /// </item>
+    /// <item>
+    /// Functions that mutate or update syntax constructs for example 'WithXXX'.  These should be on SyntaxGenerator or
+    /// some other feature specific service.
+    /// </item>
+    /// <item>
+    /// Functions that a single item when one language may allow for multiple.  For example 'GetIdentifierOfVariableDeclarator'.
```

Comment:
Should we specify for these cases that allowing multiple returns is okay? In the example VB has multiple and C# has one. It seems likely if the case is that multiple _can_ ever be returned they should be handled, but 1 and 0 items would also need to be handled.

Reply:
Yup.  That seems reasonable.